tests/foreman/longrun/test_inc_updates.py

```diff
@@ -0,0 +1,301 @@
+"""Tests for the Incremental Update feature"""
+
+from datetime import date, timedelta
+from fauxfactory import gen_alpha
+from nailgun import entity_mixins
+from nailgun.entities import (
+    ActivationKey, ContentView, ContentViewFilterRule, ContentViewVersion,
+    Errata, ErratumContentViewFilter, HostCollection, LifecycleEnvironment,
+    Organization, Repository, Subscription, System
+)
+from robottelo.api.utils import (
+    enable_rhrepo_and_fetchid, promote, upload_manifest
+)
+from robottelo.cli.contentview import ContentView as ContentViewCLI
+from robottelo.common import manifests
+from robottelo.common.constants import PRD_SETS
+from robottelo.common.decorators import run_only_on, skip_if_bug_open
+from robottelo.vm import VirtualMachine
+from unittest import TestCase
+
+
+@run_only_on('sat')
+class TestIncrementalUpdate(TestCase):
+    """Tests for the Incremental Update feature"""
+
+    @classmethod
+    def setUpClass(cls):
+        """Creates all the pre-requisites for the Incremental updates test"""
+        # Step 1 - Create a new Organization
+        cls.org = Organization(name=gen_alpha()).create()
+
+        # Step 2 - Create two life cycle environments - DEV, QE
+        cls.dev_lce = LifecycleEnvironment(
+            name='DEV',
+            organization=cls.org
+        ).create()
+        cls.qe_lce = LifecycleEnvironment(
+            name='QE',
+            prior=cls.dev_lce,
+            organization=cls.org
+        ).create()
+
+        # Step 3: Upload manifest
+        with open(manifests.clone(), 'rb') as manifest:
+            upload_manifest(cls.org.id, manifest)
+
+        # Step 4: Enable repositories - 6Server and rhel6 sat6tools
+        rhel_66_repo_id = enable_rhrepo_and_fetchid(
+            basearch=PRD_SETS['rhel_66']['arch'],
+            org_id=cls.org.id,
+            product=PRD_SETS['rhel_66']['product'],
+            repo=PRD_SETS['rhel_66']['reponame'],
+            reposet=PRD_SETS['rhel_66']['reposet'],
+            releasever=PRD_SETS['rhel_66']['releasever']
+        )
+        rhel6_sat6tools_repo_id = enable_rhrepo_and_fetchid(
+            basearch=PRD_SETS['rhel6_sat6tools']['arch'],
+            org_id=cls.org.id,
+            product=PRD_SETS['rhel6_sat6tools']['product'],
+            repo=PRD_SETS['rhel6_sat6tools']['reponame'],
+            reposet=PRD_SETS['rhel6_sat6tools']['reposet'],
+            releasever=PRD_SETS['rhel6_sat6tools']['releasever']
+        )
```

Comment:
Ok, I just realized one issue if we are planning to use the "sate6tools" from cdn directly.
a) I think for capsule, oscap, incremental updates, katello_agents
b) I believe this will be only helpful for the jenkins cdn job but not for qe-sat6-downstream as the latest content would never be available on cdn till GA.
c) I think, What we need is Build environment to be specified in jenkins and then use the latest build from SNAP-sat6tools as custom repo for sat6tools.

Reasons: 
a) oscap and puppet needs to be tested with it's corresponding build from SNAP-sat6tools and not cdn.
b) I think errata depends upon the pacakges being brought in by sat6tools, so we won't have latest build to be used if we plan to sync content from cdn.
c) Also capsule packages also needs SNAP-sat6tools and not content from cdn.
d) Otherwise we can't run these tests till the GA.